### PR TITLE
chore(.husky)configure-eslint-precommithook for test and formatting

### PR DESCRIPTION
**Pull request by shebz2023**
**what does this Pr do?**
this pr add a precommit hook to run eslint format and test

**how does this pr achieve this**
it uses husky for hooks,eslint with rules like _doublequotes_,_no unused vars_ and more

**what are the relevant pivotal tracker id**
#187364801

### CODE REVIEW

**Review by andremugabo — APPROVED** [review]
LGTM

**Review by Aldot-02 — APPROVED** [review]
LGTM

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Looks good, just Rebase. Can you also configure prettier, as you can see you're touching files that you didn't intend to change because you're changing singlequotes to double, add prettier to prevent this and other unintended changes. @shebz2023

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "tabWidth": 2,
+    "useTabs": false,
+    "semi": true,
+    "singleQuote": false,
```
Comment (SergeInGithub) [inline .prettierrc]:
```suggestion
    "singleQuote": true,
```

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Change the minor configuration below @shebz2023

**Review by SergeInGithub — COMMENTED** [review]
Looks good. Just rebase

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Pull request merged**